agent/connect/ca/aws_pca.go

```diff
@@ -0,0 +1,481 @@
+package ca
+
+// Assumptions that need to be changed:
+// 1. ECDSA is accepted everywhere. AWS only supports RSA.
+// 2. CA's are able to cross-sign. Hosted CA's don't seem to allow this.
+// 3. Certs don't require a Subject. AWS appears to require a CN.
+// 4. Validity periods can be as short as 1h. AWS minimum is 1d.
```

Comment:
I feel like we addressed all/most of these now?

Reply:
You're right, these are all taken care of by this PR and #6055. I accidentally copied these comments from another file.